src/core/src/test/java/org/apache/jmeter/threads/TestUnmodifiableJMeterVariables.java

```diff
@@ -55,34 +56,52 @@ public void testGetIteration() {
         assertThat(unmodifiables.getIteration(), CoreMatchers.is(vars.getIteration()));
     }
 
-    @Test(expected = UnsupportedOperationException.class)
+    @Test
     public void testIncIteration() {
-        unmodifiables.incIteration();
+        Assertions.assertThrows(
+                UnsupportedOperationException.class,
+                unmodifiables::incIteration
+        );
```

Comment:
It might be that `assertThrowsUnsupported(lambda)` could be helpful here so we don't repeat the same `Assertions.assertThrows` every time, but we use just `assertThrowsUnsupported(unmodifiables::incIteration)`, `assertThrowsUnsupported(() -> unmodifiables.remove("some.key"))` and so on

Reply:
Good idea